### DISCUSSION TITLE
What is the ESN number actually all about?

### DISCUSSION THREAD

**Discussion by requa3r0** [CastagnaIT/plugin.video.netflix#1023] [Q&A] [answered]
Dear CastagnaIT

First ..thanks a lot for all the work. I use your plug-in in KODI all the time. 
Mostly on a mac mini running KODI on OSX

I'm currently setting up a bunch of **Beelink GT-King Amlogic S922X** android 9 streaming boxes for friends and family.
Manly to run KODI and smart youtube etc.

If updating the COREelec to 9.2.5 (Stable) on the sd-card (latest Leia version), the kodi netflex plugin works great just out of the box. After the winvine extraction from the chrome image is complete.

It uses this ESN : NFCDCH-01-F4X05ZRJLFJ683T3EN4ZXXXXXXXXXX (last 10 are removed)

If I install latest version of net-flex on android (sideload - its not in play) on the box
It works fine, but it is restricted to SD

The generated ESN from the android app is: 
NFANDROID1-PRV-T-L3-AZW==GTKING-4445-03FF8B9C78C848D22FEFF496850628061E4BBD5B1B98476751FCE68XXXXXXXXXX  (last 10 are removed)

I read your discussion about manly concentrating on kodi 19 matrix, so I made the specific user a kodi-tester, and installed latest version of matrix from play store, and your matrix repo. Then I install the kodi plugin.

I have tried all 3 standard versions of the ESN / Widevine setting, disabled, force L3 and force L3 *ID 4445
I have also replaced with above ESN copied from the actual android app, and tried all 3 options.
I also tried wht the ESN form the COREelec 

No matter what, I get the same error. Both on latest Kodi v18.9 "Leia" and newest version of Kodi 19 "matrix"

Try Change ESN or reset ESN
Error details
Request blacklisted by key exchange service.

So. What is going on.
 
Why can we change the ESN in the first place ?

Are we not trying to cheat the net-flex server into thinking we are an approved device, to get HD? 
Or at least a working connection.

I believe a lot of people are struggling with this, and a lot of people would like to watch net-flex at full HD on various boxes. (With kodi + your plugin) and especially on this very popular box.

How is this actually done? Are we supposed to change the ESN to make it work?

I don't have a full understanding why the ESN is working in the android app, and blacklisted in your plugin. 
Is something actually broken? Can you explain what you are trying to do with the ESN replacement.

NB: Why are the ENS so long in the android app, and you seem to only use a truncated version?

Would it not be an idea to have a selection of various box types, to simulate in the ESN setting, to make the emulation of a certified android TV box a bit more easy? If this is in fact the purpose with the ESN change option.

And if we are to use super long ESN numbers, could you have a dialogue box where we can pick them up from a file.
I use the KORE remote app for the paste. But a file load would be super great ;O)

Thanks again for all your fine work

I'm very capable, when it comes to Linux and Android, and can be helpful to test any solution you may suggest.
Just let me know, and ill jump right on it.

Kind regards 
![IMG_20210103_164118](https://user-images.githubusercontent.com/16628002/103484127-bb9a3c00-4dec-11eb-9657-347261fa9e32.jpg)
![IMG_20210103_164400](https://user-images.githubusercontent.com/16628002/103484128-bc32d280-4dec-11eb-9fd9-685db3572654.jpg)
![Screenshot at 2021-01-03 17-52-22](https://user-images.githubusercontent.com/16628002/103484130-bd63ff80-4dec-11eb-9bcd-7c68607ca8d4.png)

Requa3r0

**Comment by CastagnaIT** [comment, marked as answer]
Hi unfurnately Beelink products are bad with Netflix because are not Android certified and suspect of a fake Widevine certification at least on last firmwares.
Due to this Netflix not allow the use of GT-King and GS-King products to playback on Android.

Perhaps you can play at sd on Android with the GS-King:
On Kodi 19, From addon ESN expert settings
In this exactly order: Reset ESN > select Force Widevine L3 4445 > then select Apply button.

If still show the error as blacklisted, then you can not more use Netflix on Android with this device.

Instead the GS-King on android is already know to be blacklisted and is not possible use this device.

suggest you to use CoreElec

**Reply by requa3r0** [reply]
Can I ask why not just drop the android widevine implementation, and sideload a propper one like you do on the coreelec plug in?

Then use a generic ESN that will allow the box to connect. 

Is this approach possible on the android os?
